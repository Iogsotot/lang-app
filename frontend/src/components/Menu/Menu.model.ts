export interface MenuProps {
  smt?: any,
}
