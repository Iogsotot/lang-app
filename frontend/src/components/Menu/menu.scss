@import "bulma";
@import "bulma-switch";
// bg, btn
$main-bg: #b57ad2;

//text, border
$color_extra_black: #2b292d;
$color_black: #3c3c3c;
$color_white: #ffffff;
$color_light: #c9ced6;
$color_middle: #959dbd;

// higlights:
$color_important: #6f52ed;

aside.menu {
  z-index: 10;
  position: fixed;
  top: 0;
  left: 0;
  width: 100px;
  height: 100vh;

  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: space-between;

  background-color: var(--bg-aside-menu);
  .logo-field {
    display: block;
    background-color: var(--bg-logo);
    height: 85.2px;
    padding-top: 0.5rem;
    img {
      display: block;
      margin-left: 1.375rem;
    }
  }

  .menu__list {
    list-style: none;

    .menu__item {
      margin: 1.75rem auto;
      display: flex;
      justify-content: center;
      align-items: center;
    }
  }
  .icon {
    display: block;
    margin: auto;
    width: 2.75rem;
    height: auto;

    cursor: pointer;
    transition: all 150ms linear;
    &:hover {
      opacity: 0.5;
      transition: all 150ms linear;
    }
  }

  .down-group {
    display: flex;
    flex-direction: column;
    justify-content: center;
    height: 100px;
    position: relative;
    border-top: 1px solid var(--bg-menu-border);
  }

  .btn-menu__settings {
    height: 100%;
    position: relative;
    z-index: 2;
    border: 0;
    background: transparent;
    cursor: pointer;
    outline: none;
    background-color: var(--bg-aside-menu);
  }

  .hidden__menu {
    display: flex;
    flex-direction: column;
    padding-left: 15px;
    justify-content: center;
    width: 250px;
    height: 100px;
    position: absolute;
    right: 0;
    z-index: 1;
    transform: translateX(-100%);
    transition: all 0.3s linear;
    background-color: #373b53;
    border-radius: 0px 22px 22px 0px;
    color: $color_white;
    &.active {
      transform: translateX(100%);
      transition: all 0.3s linear;
    }
    .field {
      margin-bottom: 0.25rem;
    }
    &-item {
      span {
        display: inline-block;
        width: 160px;
        margin-right: 10px;
      }
    }
  }
}
@media (max-width: 640px) {
  aside.menu {
    width: 100%;
    height: 100px;
    top: calc(100% - 100px);
    .logo-field {
      display: none;
    }
    .down-group {
      flex-direction: row-reverse;
      justify-content: end;
      margin-top: -195px;
      margin-right: 1.5rem;
      height: 100px;
    }
    .top-group {
      z-index: 50;
      width: 420px;
    }
    .menu__list {
      display: flex;
      margin-top: 9px;
      .menu__item {
        flex-direction: row;
      }
    }
    .icon,
    .fa-3x {
      display: block;
      margin: auto;
      width: 1.75rem;
      height: auto;
    }
    .fa-3x {
      font-size: 1.8em;
    }
    .hidden__menu.active {
      width: 241px;
      height: 110px;
      top: 0;
      z-index: 50;
      right: -24px;
      transform: translateY(-100%);
      border-radius: 0px;
      visibility: inherit;
    }
    .hidden__menu {
      visibility: hidden;
    }
  }
}

.fal {
  width: 44px;
  height: 44px;
  color: rgb(128, 135, 162);
}
.fal:hover {
  color: rgb(103, 109, 134);
}

@media (max-width: 930px) {
  .logo-field {
    height: 100px !important;
  }
}
