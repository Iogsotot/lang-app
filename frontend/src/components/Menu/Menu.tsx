import './menu.scss';
import { FC, useState, useEffect } from 'react';
import { Link, useLocation } from 'react-router-dom';
import { useAction } from '../../hooks/useAction';
import { useTypedSelector } from '../../hooks/useTypedSelector';
import LogoWhite from '../../assets/images/Logo_white.png';
import TextBookIcon from '../../assets/images/textbook_icon.png';
import StatsIcon from '../../assets/images/stats_icon.png';
import SettingsIcon from '../../assets/images/settings_icon.png';
import { LOCATIONS } from '../../constants';

const { textbook } = LOCATIONS;

const Menu: FC = () => {
  const location = useLocation().pathname.split('/')[1];
  const currentLocation = useLocation();
  const { showButtons, showTranslate } = useAction();
  const store = useTypedSelector(commonStore => commonStore);
  const { displayButtons, translate } = store.wordList;
  const { isLoggedIn } = store.user;
  const [openSettings, setOpenSettings] = useState(false);
  const [disabled, setDisabled] = useState(false);

  const toggleSettigns = () => {
    setOpenSettings(!openSettings);
  };

  const onChangeButtons = () => {
    showButtons(!displayButtons);
  };

  const onChangeTranslate = () => {
    showTranslate(!translate);
  };

  useEffect(() => {
    if (location !== textbook) {
      setDisabled(true);
    } else {
      setDisabled(false);
    }
  }, [location]);

  return (
    <aside className="menu">
      <div className="top-group">
        <a href="/" className="logo-field">
          <img src={LogoWhite} alt="RS Lang" />
        </a>
        <nav>
          <ul className="menu__list">
            <li className="menu__item">
              <Link to="/textbook/1/1">
                <img src={TextBookIcon} alt="textbook" className="icon" />
              </Link>
            </li>
            {
              isLoggedIn
                ? <li className="menu__item">
                  <Link to="/stats">
                    <img src={StatsIcon} alt="stats" className="icon" />
                  </Link>
                </li>
                : <></>
            }
            <li className="menu__item">
              <Link to="/sprint">
                <i className="fal fa-running fa-3x"/>
              </Link>
            </li>
            <li className="menu__item">
              <Link to="/savannah">
                <i className="fal fa-paw-claws fa-3x"/>
              </Link>
            </li>
            <li className="menu__item">
              <Link to="/puzzle">
                <i className="fal fa-puzzle-piece fa-3x"/>
              </Link>
            </li>
            <li className="menu__item">
              <Link to="/audiocall">
                <i className="fal fa-headphones-alt fa-3x"/>
              </Link>
            </li>
          </ul>
        </nav>
      </div>
      <div className="down-group">
        <button className="btn-menu__settings" onClick={toggleSettigns}>
          <img src={SettingsIcon} alt="setting" className="icon" />
        </button>
        <div className={`hidden__menu ${openSettings ? 'active' : ''}`}>
          <div className="hidden__menu-item">
            <div className="field">
              <input
                onChange={onChangeTranslate}
                checked={translate}
                disabled={disabled}
                id="switchTranslate"
                type="checkbox"
                name="switchTranslate"
                className="switch is-info"
              />
              <label htmlFor="switchTranslate">Показывать перевод</label>
            </div>
            <div className="hidden__menu-item">
              <div className="field">
                <input
                  onChange={onChangeButtons}
                  checked={displayButtons}
                  disabled={disabled}
                  id="switchButtons"
                  type="checkbox"
                  name="switchButtons"
                  className="switch is-info"
                />
                <label htmlFor="switchButtons">Показывать кнопки</label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </aside>
  );
};
export default Menu;
