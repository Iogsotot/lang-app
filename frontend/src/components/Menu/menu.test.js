import { BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import Menu from './Menu';

describe('Test suits for Menu', () => {
  it('should match with snapshot', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
      <Menu/>
      </BrowserRouter>
    )
    .toJSON();
   expect(tree).toMatchSnapshot();
   });
  });