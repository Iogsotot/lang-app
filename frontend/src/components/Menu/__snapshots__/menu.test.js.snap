// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suits for Menu should match with snapshot 1`] = `
<aside
  className="menu"
>
  <div
    className="top-group"
  >
    <div
      className="logo-field"
    >
      <img
        alt="RS Lang"
        src="./images/logo_white.png"
      />
    </div>
    <nav>
      <ul
        className="menu__list"
      >
        <li
          className="menu__item"
        >
          <a
            href="/textbook"
            onClick={[Function]}
          >
            <img
              alt="textbook"
              className="icon"
              src="./images/textbook_icon.png"
            />
          </a>
        </li>
        <li
          className="menu__item"
        >
          <a
            href="/stats"
            onClick={[Function]}
          >
            <img
              alt="stats"
              className="icon"
              src="./images/stats_icon.png"
            />
          </a>
        </li>
        <li
          className="menu__item"
        >
          <a
            href="/sprint"
            onClick={[Function]}
          >
            <img
              alt="sprint"
              className="icon"
              src="./images/game_icon.png"
            />
          </a>
        </li>
        <li
          className="menu__item"
        >
          <a
            href="/savannah"
            onClick={[Function]}
          >
            <img
              alt="savannah"
              className="icon"
              src="./images/game_icon.png"
            />
          </a>
        </li>
        <li
          className="menu__item"
        >
          <a
            href="/find-pair"
            onClick={[Function]}
          >
            <img
              alt="find-pair"
              className="icon"
              src="./images/game_icon.png"
            />
          </a>
        </li>
        <li
          className="menu__item"
        >
          <a
            href="/audiocall"
            onClick={[Function]}
          >
            <img
              alt="audiocall"
              className="icon"
              src="./images/game_icon.png"
            />
          </a>
        </li>
      </ul>
    </nav>
  </div>
  <div
    className="down-group"
  >
    <a
      href="/audiocall"
      onClick={[Function]}
    >
      <img
        alt="setting"
        className="icon"
        src="./images/settings_icon.png"
      />
    </a>
  </div>
</aside>
`;
